### PR TITLE
quicklogic: pp3: add utils scripts

### PR DESCRIPTION
**Pull request by acomodi**
This PR adds all the utils script necessary to build the arch and RR graph devices, as well as generating IO and placement constraints when running VPR.

### CODE REVIEW

File: quicklogic/common/utils/testarch_graph.py [hunk 1]
```diff
@@ -0,0 +1,404 @@
+#!/usr/bin/env python3
```
Comment (mkurc-ant) [inline quicklogic/common/utils/testarch_graph.py:1]: Hmmm, I don't recall this script being needed for QuickLogic PP3 architecture. If it is not used anywhere and is not an important utility we might remove it.

**Review by mkurc-ant — CHANGES_REQUESTED** [review]
Most of this scripts are PP3 specific - they haven't been re-used for another architecture so far. If so then we may consider moving them to `pp3/utils` instead of keeping them in `common`.

File: quicklogic/common/utils/testarch_graph.py (hunk 1 above)
Comment (lpawelcz) [reply quicklogic/common/utils/testarch_graph.py]: I don't see it being used anywhere in the codebase so I removed it

**Comment by lpawelcz** [comment]
This PR should be merged only after https://github.com/SymbiFlow/symbiflow-arch-defs/pull/2170

**Pull request merged**